packages/framework/fluid-static/api-report/fluid-static.api.md

```diff
@@ -13,15 +13,15 @@ import { IEvent } from '@fluidframework/core-interfaces';
 import { IEventProvider } from '@fluidframework/core-interfaces';
 import { IFluidLoadable } from '@fluidframework/core-interfaces';
 import { IRuntimeFactory } from '@fluidframework/container-definitions/internal';
-import { ISharedObjectKind } from '@fluidframework/shared-object-base';
+import { SharedObjectKind } from '@fluidframework/shared-object-base';
 
 // @public
 export type ContainerAttachProps<T = unknown> = T;
 
 // @public
 export interface ContainerSchema {
-    readonly dynamicObjectTypes?: readonly LoadableObjectClass[];
-    readonly initialObjects: LoadableObjectClassRecord;
+    readonly dynamicObjectTypes?: readonly SharedObjectKind[];
```

Comment:
This is the other half of the main public APi change: using SharedObjectKind instead of LoadableObjectClass.

Reply:
someone owning this space should take a look. @ChumpChief , @sashasimic, can someone take a look?

Reply:
LoadableObjectClass currently permits DataObjects, e.g. Signaler, so this change will change the functional scenarios we support.  Is that intentional, or are you only intending to change typing while retaining the same scenario support?

I'm personally OK with removing the support as it was only declared "beta", but then you'll need to follow this up with documentation updates to reflect the reduced support (e.g. https://fluidframework.com/docs/build/data-modeling/#creating-a-dynamic-object, probably some tsdoc in various places, probably others).

Reply:
We still support data objects. No change to what works, they just use an identical public API to shared objects now. I even added a utility to aqueduct to create data objects with the correct typers to with this updated API, and updated data objects tests and examples to align with it.

The public facing docs don't mention this as data objects aren't available publicly in any way that's observable different from shared objects but if we did publish a DataObject as public (for example the experimental one in our tree-react-utils), they could just use it as if it were a shared object and it would just work.

Reply:
The docs on SharedObjectKind do call out that it can be a DataObject, but only in internal facing docs.